biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -0,0 +1,1114 @@
+  use EDPftvarcon      , only    : EDPftvarcon_inst
+  
+  implicit none
+  private
+  
+  public :: PrepNutrientAquisitionBCs
+  public :: UnPackNutrientAquisitionBCs
+  public :: FluxIntoLitterPools
+
+  
+  logical, parameter :: debug  = .false. ! local debug flag
+  character(len=*), parameter, private :: sourcefile = &
+       __FILE__
+
+
+contains
+  
+  ! =====================================================================================
+
+  function GetPlantDemand(ccohort,element_id) result(plant_demand)
+
+    ! -----------------------------------------------------------------------------------
+    ! This function calculates the plant's demand for a given nutrient
+    ! based upon the need to fill its NPP demand and/or the need to
+    ! get its tissues to their ideal stoichiometry ratios.
+    ! This routine is used for informing BGC competition schemes, and
+    ! for generating synthetic upake rates and also for calculating
+    ! diagnostics.
+    !
+    ! THIS ROUTINE IS UNDERGOING MODIFICATIONS WILL CLEAN UP AFTER
+    ! A DECENT FIRST HYPOTHESIS MANIFESTS
+    ! -----------------------------------------------------------------------------------
+
+    
+    type(ed_cohort_type),intent(in) :: ccohort
+    integer,intent(in)              :: element_id      ! Should match nitrogen_element or
+    
+    ! phosphorus_element
+    real(r8)              :: plant_demand ! Nutrient demand per plant [kg]
+    real(r8)              :: plant_x      ! Total mass for element of interest [kg]
+    real(r8)              :: plant_max_x  ! Maximum mass for element of interest [kg]
+    real(r8)              :: npp_demand
+    real(r8)              :: deficit_demand
+    real(r8)              :: leaf_c, fnrt_c, sapw_c, store_c, struct_c, repro_c
+    real(r8)              :: leaf_c_trgt, fnrt_c_trgt, sapw_c_trgt, struct_c_trgt, store_c_trgt
+    real(r8)              :: leaf_dcdd,fnrt_dcdd,sapw_dcdd,store_dcdd,struct_dcdd
+    real(r8)              :: agw_c,bgw_c,agw_dcdd,bgw_dcdd
+    real(r8)              :: leaf_c_deficit
+    real(r8)              :: daily_x_demand2
+    integer               :: pft
+    real(r8)              :: dbh
+
+    real(r8), parameter :: smth_fac = 0.8_r8         ! Smoothing factor for updating
+                                                     ! demand.
+    real(r8), parameter :: init_demand_frac = 0.1_r8 ! Newly recruited plants
+                                                     ! will specify a demand
+                                                     ! based on their total nutrient
+                                                     ! because they have not history
+                                                     ! of need yet
+
+    
+    
+    pft = ccohort%pft
+    dbh = ccohort%dbh
+
+
+    ! If the cohort has not experienced a day of integration
+    ! (and thus any allocation yet), we specify demand
+    ! base purely on a fraction of its starting nutrient content
```

Comment:
```suggestion
    ! based purely on a fraction of its starting nutrient content
```

Reply:
This is assuming a fixed stoichiometry? Maybe state in comments if that's the case?